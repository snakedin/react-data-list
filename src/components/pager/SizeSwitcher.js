import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withOptions } from "../../options/withOptions";

class SizeSwitcher extends Component {

    static propTypes = {
        pageSize: PropTypes.number,
        onParamChanged: PropTypes.func
    };

    /**
     * Handle select change option
     */
    handleChange = (event) => {
        const { onParamChanged } = this.props;
        onParamChanged('pageSize', event.target.value);
    };

    render() {
        const { pageSize, options: { pageSizes } } = this.props;

        return (
            <select className="form-control" onChange={ this.handleChange } defaultValue={ pageSize }>
                {
                    pageSizes.map((value) => {
                        return <option value={ value } key={ value }>{ value }</option>
                    })
                }
            </select>
        );
    }
}

export default withOptions(SizeSwitcher);