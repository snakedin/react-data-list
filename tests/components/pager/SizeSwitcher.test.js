import React from "react";
import { render } from "@testing-library/react";
import "../../mocks/withOptionsMock";
import SizeSwitcher from "../../../src/components/pager/SizeSwitcher";

it("pager/SizeSwitcher render", () => {
  const { rerender, asFragment } = render(<SizeSwitcher pageSize={10}/>);
  expect(asFragment()).toMatchSnapshot();
});
