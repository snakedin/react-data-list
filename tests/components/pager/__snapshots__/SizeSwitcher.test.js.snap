// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pager/SizeSwitcher render 1`] = `
<DocumentFragment>
  <select
    class="form-control"
  >
    <option
      selected=""
      value="10"
    >
      10
    </option>
    <option
      value="20"
    >
      20
    </option>
    <option
      value="30"
    >
      30
    </option>
  </select>
</DocumentFragment>
`;
